### PR TITLE
Add `CompressedId64Set.isValid` type guard

### LINKED ISSUE #9446: Add typeguard for CompressedId64Set
I've run into a case where I had a `CompressedId64Set` or `Id64Arg`, I'm currently detecting if it's compressed how the internal implementation is doing, but it would be nicer if there was a type guard exposed in the `CompressedId64Set` namespace.

### PR DESCRIPTION
**Pull request by hl662**
## Why

Consumers working with loosely-typed data (e.g. deserialized JSON) often need to test whether a string value is a `CompressedId64Set` or a plain `Id64String`. There was no public API for this — callers had to copy the internal heuristic (`typeof x === "string" && x.startsWith("+")`) themselves. `DisplayStyle.deserialize` in `core/backend` was doing exactly that.

## What

- Adds `CompressedId64Set.isValid(ids: unknown): ids is CompressedId64Set` to the `CompressedId64Set` namespace in `@itwin/core-bentley`.
  - Returns `true` for an empty string (empty set) or any string starting with `"+"`.
  - Takes `unknown` so it works as a proper narrowing type guard in any context.
- Updates `DisplayStyle.deserialize` in `core/backend` to use the new guard, and replaces the adjacent `decompressSet().forEach(push)` pattern with the more direct `decompressArray`.
- Adds tests covering valid inputs, plain `Id64String`, non-string types, and edge cases.
- Adds a `NextVersion.md` entry.

## Validation

- Targeted: `rushx build && vitest --run` in `core/bentley` — 174/174 passing.
- Known baseline: `core/backend` build requires sibling packages built via `rush build`; no errors in `DisplayStyle.ts` itself.

Fixes #9446

---

*Nambot 🤖 (powered by Claude Sonnet 4.6)*

### CODE REVIEW

**Review by aruniverse — APPROVED** [review]

File: core/backend/src/DisplayStyle.ts [hunk 1]
```diff
@@ -38,12 +38,8 @@ export abstract class DisplayStyle extends DefinitionElement {
...
       const excludedElements = elProps.jsonProperties.styles.excludedElements;
-      if (typeof excludedElements === "string" && excludedElements.startsWith("+")) {
-        const ids: string[] = [];
-        CompressedId64Set.decompressSet(excludedElements).forEach((id: string) => {
-          ids.push(id);
-        });
-        elProps.jsonProperties.styles.excludedElements = ids;
+      if (CompressedId64Set.isValid(excludedElements)) {
+        elProps.jsonProperties.styles.excludedElements = CompressedId64Set.decompressArray(excludedElements);
       }
```
[the remark this answers is not in the document]
Comment (hl662) [reply core/backend/src/DisplayStyle.ts:43]: Good catch — fixed in b8707e0. Removed the redundant truthy pre-check entirely. `isValid(undefined)` already returns `false`, so the outer guard was only adding noise and incorrectly skipping the `""` (empty-set) case.

File: docs/changehistory/NextVersion.md [hunk 2]
```diff
@@ -19,6 +21,23 @@ publish: false
...
+
+```typescript
+function processIds(ids: CompressedId64Set | Id64String): void {
+  if (CompressedId64Set.isValid(ids)) {
+    for (const id of CompressedId64Set.iterable(ids))
+      doSomething(id);
+  } else {
+    doSomething(ids);
+  }
+}
```
[the remark this answers is not in the document]
Comment (hl662) [reply docs/changehistory/NextVersion.md]: Good point — fixed in b8707e0. Both `CompressedId64Set` and `Id64String` are `string` type aliases so the union collapsed and the example was misleading. Updated the parameter to `unknown` so the snippet actually demonstrates the narrowing benefit of the guard.

File: core/bentley/src/CompressedId64Set.ts [hunk 3]
```diff
@@ -24,6 +24,16 @@ export type CompressedId64Set = string;
  * @public
  */
 export namespace CompressedId64Set { // eslint-disable-line @typescript-eslint/no-redeclare
+  /** Returns `true` if `ids` is a valid [[CompressedId64Set]] — either an empty string
+   * (representing an empty set) or a non-empty string beginning with `'+'`.
+   * Useful as a type guard to distinguish a compressed set from an [[Id64String]] or other
+   * value when the type of a string is not known statically.
+   * @see [[CompressedId64Set.iterable]] to iterate the Ids represented by a compressed string.
```
Comment (aruniverse) [inline core/bentley/src/CompressedId64Set.ts:34]: The docstring says "valid" without caveating that this is a **prefix-only heuristic**, not a structural validator. Strings like `"+0"` (zero increment) and `"+garbage"` (non-hex content) pass this guard but throw `"Invalid CompressedId64Set"` when iterated.

Consider adding a `@note` before the closing `*/`:

```
 * @note This is a syntactic prefix check only — it does not validate the encoded content after `'+'`. A value may pass this guard and still throw during [[CompressedId64Set.iterator]] (e.g. `'+0'` or `'+garbage'`).
```

File: core/bentley/src/test/Id.test.ts [hunk 4]
```diff
@@ -631,6 +631,25 @@ describe("CompressedId64Set", () => {
...
+
+    // invalid: Id64String
+    expect(CompressedId64Set.isValid("0x1")).to.be.false;
+    // invalid: arbitrary string
+    expect(CompressedId64Set.isValid("hello")).to.be.false;
+    // invalid: non-string types
+    expect(CompressedId64Set.isValid(undefined)).to.be.false;
+    expect(CompressedId64Set.isValid(null)).to.be.false;
+    expect(CompressedId64Set.isValid(42)).to.be.false;
+    expect(CompressedId64Set.isValid(new Set())).to.be.false;
```
Comment (aruniverse) [inline core/bentley/src/test/Id.test.ts:651]: The suite covers common valid/invalid cases well. Consider adding a few cases that document the heuristic boundary — strings that *pass* `isValid` but throw when iterated — so future readers can't assume the two are equivalent:

```typescript
// passes isValid (prefix check) but throws during iteration
expect(CompressedId64Set.isValid("+0")).to.be.true;
expect(CompressedId64Set.isValid("+garbage")).to.be.true;
expect(() => [...CompressedId64Set.iterable("+0")]).to.throw();
expect(() => [...CompressedId64Set.iterable("+garbage")]).to.throw();
```

File: core/backend/src/DisplayStyle.ts [hunk 5]
```diff
@@ -36,14 +36,10 @@ export abstract class DisplayStyle extends DefinitionElement {
...
-      if (typeof excludedElements === "string" && excludedElements.startsWith("+")) {
-        const ids: string[] = [];
-        CompressedId64Set.decompressSet(excludedElements).forEach((id: string) => {
-          ids.push(id);
-        });
-        elProps.jsonProperties.styles.excludedElements = ids;
+    if (!displayOptions?.compressExcludedElementIds) {
+      const styles = elProps.jsonProperties?.styles;
+      if (styles && CompressedId64Set.isValid(styles.excludedElements)) {
+        styles.excludedElements = CompressedId64Set.decompressArray(styles.excludedElements);
```
Comment (aruniverse) [inline core/backend/src/DisplayStyle.ts:42]: The refactored decompression branch (compressed `excludedElements` → `decompressArray`) doesn't appear to have dedicated test coverage — the existing `DisplayStyle.test.ts` exercises the `Id64Array` path but not this one.

Consider adding a case where `excludedElements` is a `CompressedId64Set` string (e.g. `"+1+3"`) to confirm the decompression path still works after the refactor.

File: core/bentley/src/CompressedId64Set.ts (hunk 3 above)
Comment (hl662) [reply core/bentley/src/CompressedId64Set.ts:34]: Good call — added a `@note` in 8325df9 clarifying this is a syntactic prefix check only, and that values like `'+0'` or `'+garbage'` may pass the guard but still throw during iteration.

File: core/bentley/src/test/Id.test.ts (hunk 4 above)
Comment (hl662) [reply core/bentley/src/test/Id.test.ts:651]: Added boundary cases in 8325df9 — `'+0'` and `'+garbage'` both show `isValid` returning `true` followed by `iterable` throwing, so the distinction between the prefix check and structural validity is explicit in the tests.

File: core/backend/src/DisplayStyle.ts (hunk 5 above)
Comment (hl662) [reply core/backend/src/DisplayStyle.ts:42]: This path is already covered in `ExcludedElements.test.ts` — the `test(true)` call stores `'+123'` (a `CompressedId64Set`) in the iModel, then `getStyle(false)` at line 61 asserts that loading without `compressExcludedElementIds` returns the decompressed array `['0x123']`. That round-trip exercises our `isValid` + `decompressArray` branch directly.
